File: code/ch17/17.5.1.poorPigs.java
import java.lang.Math;

public class Solution {
  public int poorPigs(int buckets, int minutesToDie,
      int minutesToTest) {
    int cnt = 0;
    while (Math.pow(minutesToTest / minutesToDie + 1, cnt)
        < buckets)
      cnt += 1;
    return cnt;
  }
}

File: code/ch17/17.5.1.poorPigs.cpp
#include <cstdlib>
#include <cmath>

class Solution {
public:
  int poorPigs(int buckets, int minutesToDie,
      int minutesToTest) {
    int cnt = 0;
    while (pow(minutesToTest / minutesToDie + 1, cnt) <\
        buckets)
      cnt += 1;
    return cnt;
  }
};
